shared/devices/index.js

```diff
@@ -49,6 +54,10 @@ class Devices extends React.PureComponent<Props, State> {
     )
 
   render() {
+    if (this.props.hasNewlyRevoked && !this.state.toggledExplicitly) {
```

Comment:
don't call setState in render. just initialize the state correctly on line 35. we then don't need this toggledExplicitly state

Reply:
i had trouble with this. the toggle looked like it was opening and then immediately closing when the component first rendered. :| 
i'll give it another shot.

Reply:
ok. i think i got it.